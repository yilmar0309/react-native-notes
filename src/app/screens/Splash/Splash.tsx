import React, { useEffect } from 'react';
import { Text } from 'react-native';
import { useDispatch } from 'react-redux';
import { SafeAreaView } from 'react-native-safe-area-context';

import { Props, SlicesSplash } from '../../navigation/interfaceInject';

import useStyles from './styles';

const Splash: React.FC<Props & SlicesSplash> = (props) => {
  const { configTheme, navigateWithReset } = props;
  const styles = useStyles(configTheme);
  const dispatch = useDispatch();

  useEffect(() => {
    async function funAsync() {
      const user: any = await dispatch(props.validateAuthLocalRedux());
      if (user) {
        navigateWithReset('TabsNavigation')
      } else {
        navigateWithReset('Login')
      }
    }
    funAsync();
  }, [])

  return (
    <SafeAreaView style={styles.container}>
      <Text>Welcome</Text>
    </SafeAreaView>
  )
}

export default Splash;
