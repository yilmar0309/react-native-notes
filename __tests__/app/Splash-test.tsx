/**
 * @format
 */

import 'react-native';
import React from 'react';
import Splash from '@screens/Splash/Splash';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}));

test('renders correctly', () => {
  const configTheme = {
    primary: '#FE4A01',
    secondary: '#FEE2CE',
    backgroundScreens: '#f2f2f4',
    textLight: 'white',
    textDark: 'black',
    textPrimary: '#000000',
    textSecondary: '#B9B9BB',
    textActionPrimary: '',
    textButtonPrimary: 'white',
    textButtonSecondary: '#1e344f',
    textInputTitle: '#B9B9BB',
    textInputText: 'black',
    card: 'white',
  }
  const navigation = {
    addListener: jest.fn(),
    canGoBack: jest.fn(),
    dispatch: jest.fn(),
    getParent: jest.fn(),
    getState: jest.fn(),
    goBack: jest.fn(),
    isFocused: jest.fn(),
    navigate: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    push: jest.fn(),
    removeListener: jest.fn(),
    replace: jest.fn(),
    reset: jest.fn(),
    setOptions: jest.fn(),
    setParams: jest.fn(),
  };
  const validateAuthLocalRedux = jest.fn()
  const navigateWithReset = (route: string, params?: any) => null;
  const route = {
    key: 'Splash',
    name: 'Splash',
    params: undefined,
  };
  const testRenderer = renderer.create(<Splash 
    route={route} 
    navigation={navigation} 
    configTheme={configTheme} 
    navigateWithReset={navigateWithReset} 
    validateAuthLocalRedux={validateAuthLocalRedux}
  />)
  testRenderer.root;
});

