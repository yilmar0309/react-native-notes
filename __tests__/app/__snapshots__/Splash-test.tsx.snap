// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#f2f2f4",
      "flex": 1,
    }
  }
>
  <Text>
    Splash
  </Text>
</RNCSafeAreaView>
`;
